lib/compile.js

```diff
@@ -50,7 +50,7 @@
                     var parserGrammar = nearley.Grammar.fromCompiled(require('./nearley-language-bootstrapped.js'));
                     var parser = new nearley.Parser(parserGrammar);
                     parser.feed(f);
-                    var c = Compile(parser.results[0], {file: path, __proto__:opts});
+                    var c = Compile(parser.results[0], {file: path, alreadycompiled:[], __proto__:opts});
```

Comment:
I realise this makes your modified test pass, but why does this fix the problem? Won't this make us compile some files more than once?

Also, since you're already changing this line, I propose we use `Object.assign` instead of `__proto__` here. :)

Reply:
Just a note: `Object.assign` requires a polyfill for ES5 environments and everything in `lib` seems to be ES5 so far.

Reply:
Apologies for the late reply.

@tjvr Re your first point: Thanks for catching that issue. I've put up a new patch that will forward previously compiled rules to the recursive call to `Compile()`,  so each file will no longer be compiled more than once.

Re your second point. I had the same thought as Kaspar. But I'd be more than happy to start using  ES2015 going forward, but I think it should at least be in a separate commit that mentions ES2015, so people runner (very) outdated versions of JS engines can know what happened without having to dig through the code.